indico/modules/rb_new/controllers/backend.py

```diff
@@ -298,26 +298,38 @@ def _process(self):
 
 
 class RHCreateRoomBlocking(RHRoomBookingBase):
-    @use_kwargs({
+    @use_args({
         'room_ids': fields.List(fields.Int(), missing=[]),
         'start_date': fields.Date(),
         'end_date': fields.Date(),
         'reason': fields.Str(),
-        'allowed_principals': fields.List(fields.Dict(), missing=[])
+        'allowed_principals': fields.List(fields.Dict())
     })
-    def _process(self, room_ids, start_date, end_date, reason, allowed_principals):
-        rooms = Room.query.filter(Room.id.in_(room_ids)).all()
-        allowed = []
-        for obj in allowed_principals:
-            if obj.get('is_group'):
-                allowed.append(GroupProxy(obj['id'], provider=obj['provider']))
-            else:
-                allowed.append(User.query.filter(User.id == obj['id']).one())
-        blocking = create_blocking(rooms, start_date, end_date, reason, allowed)
+    def _process(self, args):
+        blocking = create_blocking(args)
         approve_or_request_blocking(blocking)
         return jsonify_data(flash=False)
 
 
+class RHUpdateRoomBlocking(RHRoomBookingBase):
+    def _check_access(self):
+        RHRoomBookingBase._check_access(self)
+        if not self.blocking.can_be_modified(session.user):
+            raise Forbidden
+
+    def _process_args(self):
+        self.blocking = Blocking.get_one(request.view_args['blocking_id'])
+
+    @use_args({
+        'room_ids': fields.List(fields.Int(), required=True),
+        'reason': fields.Str(required=True),
+        'allowed_principals': fields.List(fields.Dict(), missing=[])
+    })
+    def _process(self, args):
+        update_blocking(self.blocking, args)
+        return jsonify_data(flash=False, blocking=blockings_schema.dump(self.blocking, many=False).data)
```

Comment:
just use `jsonify()`; we don't include flashes anyway and the code doesn't check for a `.success` attribute (which is the only thing `jsonify_data(flash=False)` adds compared to plain `jsonify`)